src/ethereum_test_tools/common/types.py

```diff
@@ -489,13 +503,71 @@ def to_kwargs_dict(account: "Dict | Account | None") -> Dict:
         return cls(**kwargs)
 
 
+class Sender(Address):
+    """
+    Address of the sender of a transaction.
+    """
+
+    key: Hash | None
+    nonce: Number
+
+    def __new__(
+        cls,
+        address: "FixedSizeBytesConvertible | Address | Sender | None" = None,
+        *,
+        key: FixedSizeBytesConvertible | None = None,
+        nonce: NumberConvertible = 0,
+    ):
+        """
+        Init the sender.
+        """
+        if address is None:
+            if key is None:
+                raise ValueError("impossible to initialize sender without address")
```

Comment:
I tried but `isinstance(key, FixedSizeBytesConvertible)` does not really work because `FixedSizeBytesConvertible` is a type alias, so instance checking is not supported.

We could add the type checks for all the types it contains, but this is already done inside of the `Hash` function.